src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3812,6 +3871,92 @@ bool emberAfDoorLockClusterClearHolidayScheduleCallback(
     return true;
 }
 
+bool emberAfDoorLockClusterSetAliroReaderConfigCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                        const Commands::SetAliroReaderConfig::DecodableType & commandData)
+{
+    DoorLockServer::Instance().setAliroReaderConfigCommandHandler(commandObj, commandPath, commandData.signingKey,
+                                                                  commandData.verificationKey, commandData.groupIdentifier,
+                                                                  commandData.groupResolvingKey);
+    return true;
+}
+
+bool emberAfDoorLockClusterClearAliroReaderConfigCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                          const Commands::ClearAliroReaderConfig::DecodableType & commandData)
+{
+    DoorLockServer::Instance().clearAliroReaderConfigCommandHandler(commandObj, commandPath);
+    return true;
+}
+
+void DoorLockServer::setAliroReaderConfigCommandHandler(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                        const ByteSpan & signingKey, const ByteSpan & verificationKey,
+                                                        const ByteSpan & groupIdentifier,
+                                                        const Optional<ByteSpan> & groupResolvingKey)
+{
+    EndpointId endpointID = commandPath.mEndpointId;
+    ChipLogProgress(Zcl, "[SetAliroReaderConfig] Incoming command [endpointId=%d]", endpointID);
+
+    // If Aliro Provisioning feature is not supported, return UNSUPPORTED_COMMAND.
+    if (!SupportsAliroProvisioning(endpointID))
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro Provisioning is not supported [endpointId=%d]", endpointID);
+        commandObj->AddStatus(commandPath, Status::UnsupportedCommand);
+        return;
+    }
+
+    Delegate * delegate = GetDelegate(endpointID);
+    VerifyOrReturn(delegate != nullptr, ChipLogError(Zcl, "Delegate is null"));
+
+    uint8_t buffer[kAliroReaderVerificationKeySize];
+    MutableByteSpan readerVerificationKey(buffer);
+
+    // If Aliro reader verification key attribute is not null (i.e not empty), we can't set a new reader config without clearing the
+    // previous one, return INVALID_IN_STATE.
+    if (CHIP_NO_ERROR == delegate->GetAliroReaderVerificationKey(readerVerificationKey) && !readerVerificationKey.empty())
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro reader verification key is not null. Return INVALID_IN_STATE");
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return;
+    }
+
+    // If Aliro BLE UWB feature is supported and groupResolvingKey is not provided in the command, return INVALID_COMMAND.
+    if (SupportsAliroBLEUWB(endpointID) && !groupResolvingKey.HasValue())
```

Comment:
This check, per spec, should be before the readerVerificationKey check.

Reply:
done.